amivapi/events/authorization.py

```diff
@@ -15,9 +15,20 @@ def create_user_lookup_filter(self, user_id):
         return {'user': user_id}
 
     def has_item_write_permission(self, user_id, item):
-        """Users can only see their own signups, so they may change all visible
-        signups"""
-        return True
+        """Users can only see their own signups. They may change their own
+        signups only within the registration window of the event."""
+        if isinstance(item['event'], dict):
+            event = item['event']
+        else:
+            # Event is not embedded, get the event first
+            lookup = {current_app.config['ID_FIELD']: item['event']}
+            event = current_app.data.find_one('events', None, **lookup)
+        time_now = datetime.now(timezone.utc)
+
+        if time_now - event['time_register_start'] > timedelta(seconds=0) and \
+           time_now - event['time_register_end'] < timedelta(seconds=0):
```

Comment:
Why the substraction business instead of `if event['time_register_start] <= time_now <= event['time_register_end']`? (yep, the double `<` works in python btw^^)

Reply:
So actually this could be shortened to:

```
return event['time_register_start] <= time_now <= event['time_register_end']
```

Reply:
Furthermore, `datetime` includes the `utcnow` method, shortening the above to: 

```
return event['time_register_start] <= dt.utcnow() <= event['time_register_end']
```